jvm.c

```diff
@@ -1197,22 +1206,21 @@ stack_entry_t *execute(method_t *method,
             char *class_name = find_class_name_from_index(index, clazz);
             class_file_t *target_class;
 
-            /* FIXME: use linked list to prevent wasted space */
-            class_file_t **stack = malloc(sizeof(class_file_t *) * 100);
-            size_t count = 0;
-            while (true) {
+            class_file_t *list = calloc(1, sizeof(class_file_t));
+            init_list(list);
+
+            while (strcmp(class_name, "java/lang/Object")) {
                 find_or_add_class_to_heap(class_name, prefix, &target_class);
                 assert(target_class && "Failed to load class in i_new");
-                stack[count++] = target_class;
+                list_add(target_class, list);
                 class_name = find_class_name_from_index(
                     target_class->info->super_class, target_class);
-                if (!strcmp(class_name, "java/lang/Object"))
-                    break;
             }
 
-            /* call static initialization */
-            while (count) {
-                target_class = stack[--count];
+            /* reversely call static initialization if class have not been
+             * initialized */
+            list_for_each(target_class, list)
```

Comment:
Update `.clang-format` to support this macro:
```
ForEachMacros:
  - list_for_each
```

Then, `clang-format` can indent better.